src/js/providers/tracks-mixin.ts

```diff
@@ -310,8 +310,13 @@ const Tracks: TracksMixin = {
 
             for (i; i < len; i++) {
                 const track = tracks[i] as TextTrackLike;
-                let trackId: string = track._id || '';
-                if (!trackId) {
+                let trackId: string = track._id || '';                
+
+                if (hideIfOldTrackSafari(tracks, track, trackId)) {
```

Comment:
We always reuse id3 metadata tracks (`track.kind === 'metadata'`) so I'm not sure this logic belongs up here. Maybe those wouldn't have a `parsed` property on them but that isn't clear with these changes. We always want to assign a `trackId` to metadata tracks below and this change could potentially block that from happening.

Reply:
fixed!